### PR TITLE
Sk0 113 store validation in the database

### PR DESCRIPTION
**Pull request by Thomas851204**
Had some difficulties with devop related issues, thanks for the help.
Need a review.

### CODE REVIEW

**Review by rolcho — APPROVED** [review]
looking good! nice job! :sunglasses:

File: sk0-front/src/app/auth/store/auth.effects.ts [hunk 1]
```diff
@@ -60,6 +62,11 @@ export class AuthEffects {
       mergeMap((action) =>
         this.authService.register(action).pipe(
           map((response) => registerSuccess(response)),
+          tap(async () => {
+            await this.notificationService.alertSuccess(
+              "Megerősítő email elküldve. Nézze meg az email fiókját."
+            );
+          }),
```
Comment (markkovari) [inline sk0-front/src/app/auth/store/auth.effects.ts]: This is cool for now, be we definitely can separate it later into a notificationEffects too. And decouple it from the auth effects, in that case it only would listen to the `registerSuccess` effects.

File: sk0-front/src/app/auth/store/auth.reducer.ts [hunk 2]
```diff
@@ -51,23 +51,7 @@ const authStore = createReducer(
...
-    registerFailure,
-    (state, action): AuthState => ({
-      ...state,
-      isLoading: false,
-      error: action.message
-    })
-  ),
-  on(
-    registerSuccess,
-    (state, action): AuthState => ({
```
Comment (markkovari) [inline sk0-front/src/app/auth/store/auth.reducer.ts:73]: Why are these things removed? I did not found the equivalent changes in other reducers. Is it a "breaking change"?

**Review by markkovari — DISMISSED** [review]

File: sk0-front/src/app/auth/store/auth.reducer.ts (hunk 2 above)
Comment (rolcho) [reply sk0-front/src/app/auth/store/auth.reducer.ts:73]: these were doubled. you'll see if you expand

Comment (markkovari) [reply sk0-front/src/app/auth/store/auth.reducer.ts:73]: Ouch sorry, thanks for the clarifications, the pr diff did not show me. I stand corrected. 🚀

File: sk0-front/src/app/auth/store/auth.effects.ts [hunk 3]
```diff
@@ -69,6 +69,11 @@ export class AuthEffects {
       mergeMap((action) =>
         this.authService.register(action).pipe(
           map((response) => registerSuccess(response)),
+          tap(async () => {
```
Comment (rolcho) [inline sk0-front/src/app/auth/store/auth.effects.ts]: We should handle it in `handleRegisterSuccessEffects$` we already have it in dev branch. You should also refactor the test and the e2e for it.

**Review by rolcho — CHANGES_REQUESTED** [review]

File: sk0-front/src/app/auth/store/auth.effects.ts (hunk 3 above)
Comment (Thomas851204) [reply sk0-front/src/app/auth/store/auth.effects.ts]: You're right, I thought I already did that. Should be good now.

**Review by markkovari — APPROVED** [review]

**Review by rolcho — APPROVED** [review]

**Review by rolcho — APPROVED** [review]

**Review by markkovari — APPROVED** [review]

**Pull request merged**